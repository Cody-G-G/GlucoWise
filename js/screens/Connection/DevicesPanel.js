'use strict';
import React, {Component} from 'react';
import {ListItem} from 'native-base';
import {View, Text, ListView} from 'react-native';
import styles from "./styles";
import DeviceListRow from "./DeviceListRow";

export default class DevicesPanel extends Component {
    constructor(props) {
        super(props);
        this.ds = new ListView.DataSource({rowHasChanged: (r1, r2) => r1 !== r2});
    }

    render() {
        const dataSource = this.ds.cloneWithRows(this.props.scannedDevices);

        return (
            <View style={styles.devicesPanel}>
                <ListItem itemDivider>
                    <Text style={styles.deviceListHeader}>Found Devices</Text>
                </ListItem>

                <ListView dataSource={dataSource}
                          enableEmptySections={true}
                          renderRow={(rowData) =>
                                <DeviceListRow device={rowData}
                                               connectedDeviceIDs={this.props.connectedDeviceIDs}
                                               onPress={this.props.onPress}
                                               devicesTogglingConnection={this.props.devicesTogglingConnection}/>
                          }
                />
            </View>
        );
    }
}