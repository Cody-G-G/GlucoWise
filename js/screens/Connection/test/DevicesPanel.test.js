'use strict';
jest.mock('ListView', () => {
    return require('react').createClass({
        statics: {
            DataSource: require.requireActual('ListView').DataSource,
        },
        render() {
            return require('react').createElement('ListView', this.props, this.props.children);
        }
    });
});
import DevicesPanel from "../DevicesPanel";
import React from 'react';
import 'react-native';
import {ListView} from 'react-native';
import renderer from 'react-test-renderer';

test('DevicesPanel - renders correctly', () => {
    const device1 = {name: "DEVICE_NAME", id: "DEVICE_ID"};
    const device2 = {name: "DEVICE_NAME2", id: "DEVICE_ID2"};
    const device3 = {name: "DEVICE_NAME3", id: "DEVICE_ID3"};
    const scannedDevices = (new ListView.DataSource({rowHasChanged: (r1, r2) => r1 !== r2})).cloneWithRows([
        device1, device2, device3
    ]);

    const tree = renderer.create(
        <DevicesPanel scannedDevices={scannedDevices}/>
    ).toJSON();

    expect(tree).toMatchSnapshot();
});
