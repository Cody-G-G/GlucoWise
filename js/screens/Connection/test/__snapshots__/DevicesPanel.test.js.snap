// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DevicesPanel - renders correctly 1`] = `
<View
  style={
    Object {
      "alignSelf": "stretch",
      "flex": 6,
    }
  }
>
  <View
    accessibilityComponentType={undefined}
    accessibilityLabel={undefined}
    accessibilityTraits={undefined}
    accessible={true}
    hitSlop={undefined}
    isTVSelectable={true}
    onLayout={undefined}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "backgroundColor": "#ddd",
        "borderBottomWidth": 1,
        "borderColor": "#ddd",
        "flexDirection": "row",
        "height": 45,
        "justifyContent": "flex-start",
        "opacity": 1,
        "padding": 12,
      }
    }
    testID={undefined}
    tvParallaxProperties={undefined}
  >
    <View
      style={
        Object {
          "alignSelf": "stretch",
          "flex": 1,
          "justifyContent": "center",
        }
      }
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        foregroundColor={undefined}
        style={
          Object {
            "fontSize": 20,
            "fontWeight": "bold",
          }
        }
      >
        Found Devices
      </Text>
    </View>
  </View>
  <ListView
    dataSource={
      ListViewDataSource {
        "items": 3,
      }
    }
    enableEmptySections={true}
    renderRow={[Function]}
  />
</View>
`;
